src/component/contents/DraftEditorBlock.react.js

```diff
@@ -193,17 +210,30 @@ class DraftEditorBlock extends React.Component {
     }).toArray();
   }
 
+  _renderBlockMap(
+    blocks: BlockMap
+  ): React.Element {
+    var DraftEditorBlocks = this.props.DraftEditorBlocks;
+    return <DraftEditorBlocks {...this.props} blocksAsArray={blocks.valueSeq().toArray()} />;
+  }
+
   render(): React.Element<any> {
-    const {direction, offsetKey} = this.props;
+    const {direction, offsetKey, getBlockChildren, block} = this.props;
     const className = cx({
       'public/DraftStyleDefault/block': true,
       'public/DraftStyleDefault/ltr': direction === 'LTR',
       'public/DraftStyleDefault/rtl': direction === 'RTL',
     });
 
+    const nestedBlocks = getBlockChildren ? getBlockChildren(block.getKey()) : [];
+
+    // Render nested blocks or text but never both at the same time.
     return (
       <div data-offset-key={offsetKey} className={className}>
-        {this._renderChildren()}
+        {nestedBlocks && nestedBlocks.size && nestedBlocks.size > 0 ?
```

Comment:
Can these checks extracted into a function? I spotted we do a similar check in `shouldComponentUpdate`.